src/components/content-editor/content-accordion.tsx

```diff
@@ -0,0 +1,539 @@
+        if (pageLayoutView.widget === PageLayoutWidgetType.MaterialTabLayout) {
+          elementList.push(renderAndroidTabs(pageLayoutView));
+        } else {
+          elementList.push(renderAndroidResources(selectedPage, pageLayoutView, idList));
+        }
+      }
+
+      if (pageLayoutView.children.length > 0) {
+        renderAndroidViews(elementList, pageLayoutView.children);
+      }
+    });
+
+    return elementList;
+  };
+
+  /**
+   * Returns list of HTML components as flat list
+   * 
+   * @param parentComponent parent component
+   * @returns child components as flat list
+   */
+  const getHtmlComponentList = (parentComponent: TreeObject): TreeObject[] => {
+    const result: TreeObject[] = [parentComponent];
+
+    parentComponent.children.forEach((child) => {
+      result.push(...getHtmlComponentList(child));
+    });
+
+    return result;
+  };
+
+  /**
+   * Renders HTML layout component editor
+   * 
+   * @param component component
+   */
+  const renderHtmlLayoutComponent = (component: TreeObject) => {
+    const componentResourceIds = HtmlResourceUtils.getTreeObjectResourceIds(component);
+    const resources = selectedPage.resources.filter((resource) => componentResourceIds.includes(resource.id));
+    const eventTriggerItems = null;
+
+    return (
+      <Accordion
+        key={component.id}
+        className={classes.resource}
+      >
+        <AccordionSummary expandIcon={<ExpandMoreIcon />} className={classes.resourceItem}>
+          <Typography variant="h5">{component.name || ""}</Typography>
+          <Button variant="text" onClick={onAddEventTriggerClick(component.id)}>
+            {strings.contentEditor.editor.eventTriggers.add}
+          </Button>
+        </AccordionSummary>
+        <AccordionDetails>
+          <LayoutViewResourcesList
+            resources={resources}
+            selectedResource={selectedResource}
+            onClick={onResourceClick}
+          />
+          {eventTriggerItems && (
+            <div style={{ marginLeft: theme.spacing(4) }}>
+              <Typography style={{ padding: theme.spacing(1) }} variant="h5">
+                {strings.contentEditor.editor.eventTriggers.title}
+              </Typography>
+              {eventTriggerItems}
+            </div>
+          )}
+        </AccordionDetails>
+      </Accordion>
+    );
+  };
+
+  /**
+   * Renders page elements
+   */
+  const renderPageElements = () => {
+    if (selectedPageLayout.layoutType === LayoutType.Android) {
+      const elementList: JSX.Element[] = [];
+      return renderAndroidViews(elementList, [selectedPageLayout.data as PageLayoutView]);
+    } else {
+      const componentTree = constructTree((selectedPageLayout.data as PageLayoutViewHtml).html);
+      const componentList: TreeObject[] = [];
+
+      componentTree.forEach((rootComponent) => {
+        componentList.push(...getHtmlComponentList(rootComponent));
+      });
+
+      return componentList.map(renderHtmlLayoutComponent);
```

Comment:
Filter elements that cannot have resources out of the list.

Reply:
If you mean e.g. layouts, then it would be impossible to add triggers to them, so this would probably not be a good idea.